src/quo2/components/channel_avatar.cljs

```diff
@@ -0,0 +1,47 @@
+(ns quo2.components.channel-avatar
+  (:require [quo.design-system.colors :as colors]
+            [quo2.foundations.colors :as f-colors]
+            [quo.react-native :as rn]
+            [clojure.string :as clj-string]
+            [status-im.ui.components.icons.icons :as icons]))
+
+(defn channel-avatar [{:keys [big? dark? lock-status icon x y]}]
+  [rn/view {:style {:width (if big? 32 24)
+                    :height (if big? 32 24)
+                    :top (if (or (clj-string/blank? x)
+                                 (js/isNaN x))
+                           0
+                           (js/parseInt x))
```

Comment:
Clojure way 🙂

`:top (if (int? x) x 0)`

Reply:
Still, properties like positioning of view/component will be handled while using it. So we can just pass another key `style` and merge with this predefined style ([Ex.](https://github.com/status-im/status-react/blob/578e542a6637011996166d2628781ce5fc26dc25/src/quo2/components/text.cljs#L32)). This way we will have control to overwrite style instead of passing x & y.

Reply:
Oh sorry, I missed that part.